packages/ingest/test/test-recursion.js

```diff
@@ -4,113 +4,175 @@ const test = require('ava');
+    ]);
+  });
+
+test('recursion lists all files in a simple relative text path', async (t) => {
+  const dirs = {
+    '.': [
+      { type: 'd', name: 'path' }
     ],
-    dir: [
+    './path': [
+      { type: '-', name: 'file1' },
+      { type: '-', name: 'file2' },
       { type: '-', name: 'file3' }
     ]
   };
   const fn = (path) => dirs[path];
-  const files = await recursion(fn, '');
-  t.deepEqual(files, [
-    { type: '-', name: 'file3' },
-    { type: '-', name: 'file1' },
-    { type: '-', name: 'file2' }
-  ]);
+  const files = await recursion(fn, 'path');
+  t.deepEqual(files, dirs['./path']);
 });
 
-test('recursion lists all files in a simple directory structure with text path', async (t) => {
+test('recursion lists all files in a simple absolute text path', async (t) => {
   const dirs = {
-    '': [
+    '/': [
       { type: 'd', name: 'path' }
     ],
-    path: [
+    '/path': [
       { type: '-', name: 'file1' },
       { type: '-', name: 'file2' },
       { type: '-', name: 'file3' }
     ]
   };
   const fn = (path) => dirs[path];
-  const files = await recursion(fn, 'path');
-  t.deepEqual(files, dirs.path);
+  const files = await recursion(fn, '/path');
+  t.deepEqual(files, dirs['/path']);
 });
 
-test('recursion lists all files in a simple directory structure with regex path', async (t) => {
+test('recursion lists all files in a simple relative regex path', async (t) => {
   const dirs = {
-    '': [
+    '.': [
       { type: 'd', name: 'dir1' },
       { type: 'd', name: 'dir2' }
     ],
-    dir1: [
+    './dir1': [
       { type: '-', name: 'file1' },
       { type: '-', name: 'file2' },
       { type: '-', name: 'file3' }
     ],
-    dir2: [
+    './dir2': [
+      { type: '-', name: 'file4' },
+      { type: '-', name: 'file5' },
+      { type: '-', name: 'file6' }
+    ]
+  };
+  const fn = (path) => dirs[path];
+  const files = await recursion(fn, '(dir.*)');
+  t.deepEqual(files, [...dirs['./dir1'], ...dirs['./dir2']]);
+});
+
+test('recursion lists all files in a simple absolute regex path', async (t) => {
+  const dirs = {
+    '/': [
+      { type: 'd', name: 'dir1' },
+      { type: 'd', name: 'dir2' }
+    ],
+    '/dir1': [
+      { type: '-', name: 'file1' },
+      { type: '-', name: 'file2' },
+      { type: '-', name: 'file3' }
+    ],
+    '/dir2': [
       { type: '-', name: 'file4' },
       { type: '-', name: 'file5' },
       { type: '-', name: 'file6' }
     ]
   };
   const fn = (path) => dirs[path];
   const files = await recursion(fn, '/(dir.*)');
-  t.deepEqual(files, [...dirs.dir1, ...dirs.dir2]);
+  t.deepEqual(files, [...dirs['/dir1'], ...dirs['/dir2']]);
 });
 
-test('recursion lists all files in a complex directory structure with text path', async (t) => {
+test('recursion lists all files in a complex relative text path', async (t) => {
   const dirs = {
-    '': [
+    '.': [
       { type: 'd', name: 'path' }
     ],
-    path: [
+    './path': [
```

Comment:
Same comment about including files in the parent directories in the test fixtures

Reply:
f92ab5faa